swupd-inspector/archive.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"fmt"
+	"io"
+
+	"github.com/clearlinux/mixer-tools/swupd"
+)
+
+// TODO: Evaluate where to put this, partially duplicate with code in packs.
```

Comment:
Partially? Can't the swupd library do this already? I would much prefer you extend the existing capabilities rather than duplicating much of the implementation into another package in the same project.

Reply:
Done.